### PR TITLE
bugfix - use lambdas as middleware

### PR DESCRIPTION
**Pull request by joefitter**
If a child controller overrides the render method, lambdas is no longer available unless super is called (which is often not desired). Due to the nature of lambdas it should have been implemented as a middleware, as per template mixins.

* added next param to lambdas
* call next after res.locals extended with new lambda
* remove render method from base-controller
* apply middleware with controller.use interface

### CODE REVIEW

File: lib/middleware/lambdas.js [hunk 1]
```diff
@@ -20,4 +20,5 @@ module.exports = function lambdas(req, res) {
       };
     }
   });
+  next();
 };
```
Comment (JoeChapman) [inline lib/middleware/lambdas.js:24]: Use typical express middleware signature so it's open for options and for consistency's sake

```
function (options) {
   return function (req, res, next) {
       Object.assign({
       ......
       })
       next();
   });
})
```

Comment (joefitter) [reply lib/middleware/lambdas.js:24]: this is typical - we don't always need options, we only include options if the middleware needs options - have a look at the middleware in wizard

Comment (JoeChapman) [reply lib/middleware/lambdas.js:24]: I'm looking at our middlewares

Comment (joefitter) [reply lib/middleware/lambdas.js:24]: https://github.com/UKHomeOfficeForms/hof-form-wizard/blob/master/lib/middleware/session.js

we include options if we need to, I don't see the point of defining options and doing nothing with it - that would actually fail linting

Comment (JoeChapman) [reply lib/middleware/lambdas.js:24]: Fair, I just like the consistency it offers.

Comment (joefitter) [reply lib/middleware/lambdas.js:24]: @easternbloc @lennym what say you?

Comment (lennym) [reply lib/middleware/lambdas.js:24]: I'd go for a function that returns middleware, even with no options as it provides greatest flexibility for future change without breaking.

Comment (joefitter) [reply lib/middleware/lambdas.js:24]: outnumbered, beaten into submission.... _backs down_

Comment (lennym) [reply lib/middleware/lambdas.js:24]: It's not a hugely strong preference, I don't really GAF, but you asked...

Comment (joefitter) [reply lib/middleware/lambdas.js:24]: I did! I'm changing it now

**Comment by joefitter** [comment]
@JoeChapman @lennym updated! can I get a 👍

**Review by JoeChapman — APPROVED** [review]

**Review by lennym — APPROVED** [review]

**Pull request merged**